### PR TITLE
PRMT-3486

### PR DESCRIPTION
**Pull request by martin-nhs**

### CODE REVIEW

File: src/main/java/uk/nhs/prm/e2etests/mesh/MeshClient.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        return getMessageIdFromMessage(response.body());
+    }
+
+    private String getMessageIdFromMessage(String response) {
+        return gson
+                .fromJson(response, NemsEventMessage.class)
+                .getMessage();
+    }
+
+//    private String getMessageIdFromMessage(String responseBody) {
```
Comment (AndyFlintNHS) [inline src/main/java/uk/nhs/prm/e2etests/mesh/MeshClient.java:73]: Remove this commented out method please

File: src/test/java/uk/nhs/prm/e2etests/performance/PerformanceTest.java [hunk 2]
```diff
@@ -167,15 +165,15 @@ private boolean before(LocalDateTime timeout) {
...
-            var authStrategyWarning = "Performance test timeout is approaching an hour, getting where this will not work if " +
-                    "using temporary credentials (such as obtained by user using MFA) if it exceeds the expiration time. " +
-                    "Longer runs will need to be done in pipeline where refresh can be made from the AWS instance's " +
-                    "metadata credentials lookup.";
-            System.err.println(authStrategyWarning);
-        }
-        out.println("AUTH STRATEGY: using auto-refresh, role-assuming sqs client");
-        return context.getBean(AutoRefreshingRoleAssumingSqsClient.class);
-    }
+//    private AutoRefreshingRoleAssumingSqsClient appropriateAuthenticationSqsClient() {
```
Comment (AndyFlintNHS) [inline src/test/java/uk/nhs/prm/e2etests/performance/PerformanceTest.java:168]: Can we please remove this unused method?

File: src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java [hunk 3]
```diff
@@ -131,19 +125,51 @@ private void assertMessagesAreInTransferCompleteQueue(int numberOfMessagesToBePr
...
+                smallEhr = getSmallMessageWithUniqueConversationIdAndMessageId(messageTemplate, conversationId);
+                messagesToBeProcessed.get(i).start();
 
                 System.out.println("Item " + counter.get() + " - sending to mq conversationId " + conversationId);
                 inboundQueueFromMhs.sendMessage(smallEhr, conversationId);
+
                 sleeper.sleep(intervalBetweenMessagesSentToMq);
-            });
+            }
+//            messagesToBeProcessed.forEach(message -> {
```
Comment (AndyFlintNHS) [inline src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java:146]: Can we please remove this commented out block of code? We can try go and fix this later but for now we don't want it in main

File: src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java [hunk 4]
```diff
@@ -184,15 +230,15 @@ private String getSmallMessageWithUniqueConversationIdAndMessageId(String messag
...
-            var authStrategyWarning = "Performance test timeout is approaching an hour, getting where this will not work if " +
-                    "using temporary credentials (such as obtained by user using MFA) if it exceeds the expiration time. " +
-                    "Longer runs will need to be done in pipeline where refresh can be made from the AWS instance's " +
-                    "metadata credentials lookup.";
-            System.err.println(authStrategyWarning);
-        }
-        out.println("AUTH STRATEGY: using auto-refresh, role-assuming sqs client");
-        return context.getBean(AutoRefreshingRoleAssumingSqsClient.class);
-    }
+//    private AutoRefreshingRoleAssumingSqsClient appropriateAuthenticationSqsClient() {
```
Comment (AndyFlintNHS) [inline src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java:233]: Please remove this commented out method

File: src/test/java/uk/nhs/prm/e2etests/tests/RepositoryE2ETests.java [hunk 5]
```diff
@@ -191,6 +190,7 @@ void shouldVerifyThatASmallEhrXMLIsUnchanged() {
         assertFalse(myDiff.toString(), myDiff.hasDifferences());
     }
 
+    // THIS TEST WAS FAILING BEFORE REFACTOR
```
Comment (AndyFlintNHS) [inline src/test/java/uk/nhs/prm/e2etests/tests/RepositoryE2ETests.java:193]: Mark as a todo?

**Review by AndyFlintNHS — CHANGES_REQUESTED** [review]

File: src/test/java/uk/nhs/prm/e2etests/tests/ContinuityE2E.java [hunk 6]
```diff
@@ -252,13 +239,13 @@ public void shouldDeleteEhrOfPatientOnTheirReRegistration() throws Exception {
 
 
 
-        await().atMost(20, TimeUnit.SECONDS).untilAsserted(() -> { // check it's a 404?
-            assertThrows(HttpClientErrorException.class, () -> {
-                ehrRepoClient.getEhrResponse(patientNhsNumber);
-            });
+
+        assertThrows(HttpClientErrorException.class, () -> {
+            ehrRepoClient.getEhrResponse(patientNhsNumber);
```
Comment (MohammadIqbalAD-NHS) [inline src/test/java/uk/nhs/prm/e2etests/tests/ContinuityE2E.java:244]: I believe we should revert to the original. This test was always failing even after the change.

File: src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java [hunk 7]
```diff
@@ -183,16 +203,4 @@ private String getSmallMessageWithUniqueConversationIdAndMessageId(String messag
...
-        if (config.performanceTestTimeout() > TestConfiguration.SECONDS_IN_AN_HOUR * 0.9) {
-            var authStrategyWarning = "Performance test timeout is approaching an hour, getting where this will not work if " +
-                    "using temporary credentials (such as obtained by user using MFA) if it exceeds the expiration time. " +
-                    "Longer runs will need to be done in pipeline where refresh can be made from the AWS instance's " +
-                    "metadata credentials lookup.";
-            System.err.println(authStrategyWarning);
-        }
-        out.println("AUTH STRATEGY: using auto-refresh, role-assuming sqs client");
-        return context.getBean(AutoRefreshingRoleAssumingSqsClient.class);
-    }
```
Comment (MohammadIqbalAD-NHS) [inline src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java:197]: Has this been replaced by something else? I wonder if this is necessary when running e2e tests on GoCD, as tests may take a long time to finish, which is maybe why they have this credential refreshing logic?

File: src/main/resources/application.yml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+          mqAppUsername: /repo/${NHS_ENVIRONMENT}/user-input/mq-app-username
+          mqAppPassword: /repo/${NHS_ENVIRONMENT}/user-input/mq-app-password
+        safeListedPatients: /repo/${NHS_ENVIRONMENT}/user-input/external/safe-listed-patients-nhs-numbers
+
+    # TODO: PRMT-3468 - MAKE USE OF GETENVSUFFIX, NHS_ENV HAS ONLY PROD AND NON PROD
+    # TODO: PRMT-3468 - MAKE USE OF OUR AWS ENVIRONMENTS.
+    serviceUrls:
+      pdsAdaptor: https://pds-adaptor.${NHS_ENVIRONMENT}.non-prod.patient-deductions.nhs.uk/
+      gp2GpMessenger: https://gp2gp-messenger.${NHS_ENVIRONMENT}.non-prod.patient-deductions.nhs.uk/
+      ehrRepository: https://ehr-repo.${NHS_ENVIRONMENT}.non-prod.patient-deductions.nhs.uk/
```
Comment (MohammadIqbalAD-NHS) [inline src/main/resources/application.yml:36]: Some TODO Comments Need Addressing.

File: src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java (hunk 7 above)
Comment (martin-nhs) [reply src/test/java/uk/nhs/prm/e2etests/performance/RepoInPerformanceTest.java:197]: TODO within refactoring ticket

File: src/main/java/uk/nhs/prm/e2etests/suspensions/SuspensionMessageRealQueue.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+@Component
+public class SuspensionMessageRealQueue extends QueueMessageHelper {
+
+    @Autowired
+    public SuspensionMessageRealQueue(
+            ThinlyWrappedSqsClient thinlyWrappedSqsClient,
+            QueuePropertySource queuePropertySource
+    ) {
+        // TODO: IS THIS THE CORRECT QUEUE URL
+        super(thinlyWrappedSqsClient, queuePropertySource.getSuspensionsServiceSuspensionsQueueUrl());
```
Comment (MohammadIqbalAD-NHS) [inline src/main/java/uk/nhs/prm/e2etests/suspensions/SuspensionMessageRealQueue.java:19]: Comment left Unanswered

File: src/main/java/uk/nhs/prm/e2etests/queue/QueueMessageHelper.java [hunk 10]
```diff
@@ -23,6 +24,7 @@ public class QueueMessageHelper {
     public QueueMessageHelper(ThinlyWrappedSqsClient thinlyWrappedSqsClient, String queueUri) {
         this.thinlyWrappedSqsClient = thinlyWrappedSqsClient;
         this.queueUri = queueUri;
+        log.info("======================================== {} =====================================", queueUri);
```
Comment (MohammadIqbalAD-NHS) [inline src/main/java/uk/nhs/prm/e2etests/queue/QueueMessageHelper.java:27]: Needs removing.

File: src/main/resources/application.yml (hunk 8 above)
Comment (martin-nhs) [reply src/main/resources/application.yml:36]: Parked in favour of PRMT-3573

File: src/main/java/uk/nhs/prm/e2etests/E2eTestsApplication.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.prm.e2etests;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceTransactionManagerAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+
+// TODO: Temporarily disabled Hibernate / Spring Data JPA AutoConfiguration.
```
Comment (MohammadIqbalAD-NHS) [inline src/main/java/uk/nhs/prm/e2etests/E2eTestsApplication.java:9]: Todo comment

**Pull request merged**